elasticsearch/manifests/fix-hosts.yaml

```
@@ -0,0 +1,21 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: elasticsearch-master
+spec:
+  template:
+    spec:
+      containers:
+        - name: elasticsearch
+          env:
+            - name: MY_POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: ELASTICSEARCH_CLUSTER_HOSTS
+              value:  "elasticsearch-master-hl,"
+            - name: ELASTICSEARCH_ADVERTISED_HOSTNAME
+              value: "$(MY_POD_NAME).elasticsearch-master-hl"
```

I think it might be helpful to have some more context here (i.e. replacing the comment just before including this patch in `kustomization.yaml`)

```suggestion
            - name: MY_POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            # update some vars that by default use the namespace passed to helm,
            # since we don't pass a namespace when generating these.
            # the `MY_POD_NAME` var is just following the approach used in the generated manifests
            - name: ELASTICSEARCH_CLUSTER_HOSTS
              value:  "elasticsearch-master-hl,"
            - name: ELASTICSEARCH_ADVERTISED_HOSTNAME
              value: "$(MY_POD_NAME).elasticsearch-master-hl"
```

---

I'll improve the comment in the next PR